File: medium/2348. Number of Zero-Filled Subarrays.js
// 2348. Number of Zero-Filled Subarrays

// https://leetcode.com/contest/biweekly-contest-83/problems/number-of-zero-filled-subarrays/

// #1
const zeroFilledSubarray = function(nums) {
  let temp = 0
  let count = 0


  for (let num of nums) {
    if (num === 0) {
      temp++
      count += temp
    } else {
      temp = 0
    }
  }
  return count
};

File: medium/2348. Number of Zero-Filled Subarrays.py
# 2348. Number of Zero-Filled Subarrays

# 1
class Solution:
  def zeroFilledSubarray(self, nums: List[int]) -> int:
    cnt = temp = 0

    for num in nums:
      if num == 0:
        temp += 1
        cnt += temp
      else:
        temp = 0
    return cnt